### PR TITLE
Add BIP32 path restrictions

### PR DESCRIPTION
**Pull request by tomsmalley**

### CODE REVIEW

**Comment by tomsmalley** [comment]
I missed the CI failures - fixing now.

File: tests/generative-sign-hash-tests.js [hunk 1]
```diff
@@ -9,13 +9,13 @@ const bip = require('bip32-path');
 const prefix = bip.fromString("m/44'/9000'").toPathArray();
 const bipNotHardened = fc.integer(0x7fffffff);
 const bipHardened = bipNotHardened.map(a => a + 0x80000000);
-const accountGen = bipHardened.map(a => bip.fromPathArray(prefix.concat([a])));
-const subAddressGen = fc.array(fc.integer(0,4294967295),2,2).map(bip.fromPathArray);
+const account = bip.fromPathArray(prefix.concat([0 + 0x80000000]));
```
Comment (matthewbauer) [inline tests/generative-sign-hash-tests.js:12]: Is the path here right? Or does it need to increment like bipHardened?

**Comment by matthewbauer** [comment]
There's also a bip32 suffix used in the app. Are there any restriction to enforce there?

**Review by jonored — APPROVED** [review]
LGTM with the note that this restricts the ledger app to only using the first account. We'll revisit later.

**Pull request merged**